src/main/java/seedu/address/model/student/PredicateList.java

```diff
@@ -13,8 +16,28 @@ public PredicateList(List<Predicate<Student>> predicates) {
         this.predicates = predicates;
     }
 
+    public PredicateList() {
+        predicates = new LinkedList<>();
+    }
+
+    @SafeVarargs
+    public static PredicateList of(Predicate<Student>... predicates) {
+        return new PredicateList(Arrays.asList(predicates));
+    }
+
+    public boolean add(Predicate<Student> predicate) {
+        return predicates.add(predicate);
+    }
+
+    public boolean isEmpty() {
+        return predicates.isEmpty();
+    }
+
     @Override
     public boolean test(Student student) {
+        if (predicates.isEmpty()) {
+            return false;
+        }
```

Comment:
Isn't this already handled by the FindCommandParser class?

Reply:
Added that to prevent over-coupling with the parser